### PR TITLE
Added Support for Textarea and documentation for other input types 

### PR DESCRIPTION
**Pull request by Data-Meister**
It turns out your logic works perfectly for textarea and input types like number

### CODE REVIEW

**Comment by nkbt** [comment]
Hi @Data-Meister!

Thanks for using react-debounce-input and really appreciate your PR!

I have couple of notes.

Instead of adding `type` field which is a bit ambiguous (there is no such thing as `<input type="textarea">`) I'd better use something more generic like `element`. I do it in some of my components like:

```jsx
const Button = ({element = 'button', ...props}) =>
  React.createElement(element, {...props});

Button.propTypes = {
  element: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.func])
};
```

In this way you don't need to have special cases, and also will not override any of default input props.

**Comment by Data-Meister** [comment]
That sounds good, I will implement as you suggest soon.

**Comment by Data-Meister** [comment]
OK - I pushed another commit

File: README.md [hunk 1]
```diff
@@ -82,6 +81,60 @@ ReactDOM.render(<App />, appRoot);
...
+```
+
+Note: when rendering a `<textarea />` you may wish to set `forceNotifyByEnter = { false }` so the user can make new lines without forcing notification of the current value.
+
+This package has only been tested with `<input />` and `<textarea />` but should work with any element which has `value` and `onChange` props.
+
+You can also use a custom react component as the element. For Example:
+
+```js
+<DebounceInput
```
Comment (nkbt) [inline README.md]: Wow, that is a great side-effect! Basically fixes https://github.com/nkbt/react-debounce-input/issues/5

Comment (Data-Meister) [inline README.md]: Great!

**Comment by nkbt** [comment]
Merged with some minor fixes. Also added one more example.
Published 2.4.0

**Pull request closed without merging**